File: 010-bis.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

#define YES 1
#define NO 0

int main(int argc, char *argv[]) {
    int limit=2000000;
    int i,m;
    int crosslimit=sqrt(2000000);
    char *sieve=(char *)malloc(sizeof(char)*limit); 
    double sum;
    
    for (i=0;i<limit;i++) 
        sieve[i]=NO;
    
    for (i=4;i<limit;i+=2) {
        sieve[i]=1;
    }
    
    for (i=3; i<crosslimit; i+=2) {
        if (!sieve[i]) {
            for ( m=i*i ; m<limit; m+=2*i ) {
                sieve[m]=YES;
            }
        }
    }
    sum=0;
    for (i=2;i<limit;i++) {
        if (!sieve[i]) { 
            sum += i; 
        }
    }
    printf("%lf\n",sum);
}

File: old-ruby/010-bis.rb
limit=2000000;

crosslimit=Math.sqrt(2000000);
sieve=[0]*limit;

i=0;
while i<limit
    sieve[i]=false
    i+=1
end

i=4
while i<limit
    sieve[i]=1;
    i+=2
end
  
i=3
while i<crosslimit
    if !sieve[i]
        m=i*i
        while m<limit
            sieve[m]=true;
            m+=2*i
        end
    end
    i+=2
end
sum=0;
i=2;

while i<limit
    if not sieve[i]
        sum += i
    end 
    i+=1
end
puts sum
